pkg/internal/reporter/reporter_test.go

```diff
@@ -108,47 +109,43 @@ func TestReportShouldSuccessfullyReportResults(t *testing.T) {
 	})
 
 	t.Run("on checkup failure", func(t *testing.T) {
-		fakeClient := fake.NewSimpleClientset(newConfigMap())
-		testReporter := reporter.New(fakeClient, testNamespace, testConfigMapName)
-
-		var checkupStatus status.Status
-		checkupStatus.StartTimestamp = time.Now()
-		assert.NoError(t, testReporter.Report(checkupStatus))
-
-		checkupStatus.FailureReason = []string{failureReason1}
-		checkupStatus.CompletionTimestamp = time.Now()
-		assert.NoError(t, testReporter.Report(checkupStatus))
+		const (
+			failureReason1 = "some reason"
+			failureReason2 = "some other reason"
+		)
 
-		expectedReportData := map[string]string{
-			"status.succeeded":           strconv.FormatBool(false),
-			"status.failureReason":       failureReason1,
-			"status.startTimestamp":      timestamp(checkupStatus.StartTimestamp),
-			"status.completionTimestamp": timestamp(checkupStatus.CompletionTimestamp),
+		testCases := []checkupFailureCase{
+			{
+				description:    "with no results",
+				failureReasons: []string{failureReason1},
+			},
+			{
+				description:    "with no results and multiple failures",
+				failureReasons: []string{failureReason1, failureReason2},
+			},
```

Comment:
Could you please add a test case with results and a single failure?

Reply:
it's in the last commit.